File: FindRestaurants-frontend/src/app/services/restaurant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Restaurant } from '../interface/Restaurant'
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
//This service is used for communication with the Restaurant api microservice
export class RestaurantService {
  //default RestController path
  private readonly path = `http://localhost:8080/api/restaurants`;
  constructor(private http: HttpClient) { }

  //Return all restaurant from the repository
  findAllRestaurants(): Observable<Restaurant[]> {
    return this.http.get<Restaurant[]>(`${this.path}`);
  }

  //input parametar id: the id of the Restaurant
  //returns a specific restaurant by id
  findById(id: String): Observable<Restaurant> {
    return this.http.get<Restaurant>(`${this.path}/${id}`);
  }

}

File: FindRestaurants-backend/src/main/java/com/example/FindRestaurantsbackend/service/RestaurantService.java
package com.example.FindRestaurantsbackend.service;

import com.example.FindRestaurantsbackend.domain.Restaurant;
import com.example.FindRestaurantsbackend.repository.RestaurantRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class RestaurantService {
    private final RestaurantRepository repository;

    public RestaurantService(RestaurantRepository repository) {
        this.repository = repository;
    }

    //returns a list of Restaurants from the repository
    public List<Restaurant> findAllRestaurants() {
        return repository.findAll();
    }

    //return a specific Restaurant by id
    public Optional<Restaurant> findById(Integer id) {
        return repository.findById(id);
    }
}
